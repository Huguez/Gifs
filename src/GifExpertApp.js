import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


export const GifExpertApp = ({ defaultState = [] })=> {
    
    const [ categorias, setCategoria ] = useState( defaultState )

    return (
        <div>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategoria } />
            <hr />
            <ol>
                { categorias.map( (cate, i ) => { 
                    return <GifGrid key={`${cate}`} category={ cate } />
                } ) }
            </ol>
        </div>
    );
}