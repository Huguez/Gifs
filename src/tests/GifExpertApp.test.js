import React from 'react';

import "@testing-library/jest-dom";

import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';


configure({adapter: new Adapter()});

describe('Pruebas con GifExpertApp.js', () => {

    let wrapper = shallow( <GifExpertApp/> );

    beforeEach( () => {
        wrapper = shallow( <GifExpertApp/> );
    } );

    test( 'Debe mostrar el componente correctamente', ()=>{
        const cate = [ 'taco', 'chorizo', 'empanada' ]

        let wrapper = shallow( <GifExpertApp defaultState={ cate } /> );
        
        expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find( 'GifGrid' ).length ).toBe( cate.length );
        
    } );

} );