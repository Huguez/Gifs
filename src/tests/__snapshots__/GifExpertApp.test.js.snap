// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con GifExpertApp.js Debe mostrar el componente correctamente 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="taco"
      key="taco"
    />
    <GifGrid
      category="chorizo"
      key="chorizo"
    />
    <GifGrid
      category="empanada"
      key="empanada"
    />
  </ol>
</div>
`;
